File: test/account/domain/AccountMother.ts
import {Account} from "../../../src/account/domain/Account";
import {AccountIdMother} from "./AccountIdMother";
import {CustomerIdMother} from "../../customer/domain/CustomerIdMother";
import {DateValueObjectMother} from "../../shared/domain/DateValueObjectMother";
import {DebitMother} from "./DebitMother";
import {AccountStatus} from "../../../src/account/domain/AccountStatus";

export class AccountMother {

    static withZeroBalance(): Account {
        return new Account(
            AccountIdMother.random(),
            CustomerIdMother.random(),
            AccountStatus.OPEN,
            DateValueObjectMother.random(),
            [],
            []
        );
    }

    static withThisDebit(value: number) {
        return new Account(
            AccountIdMother.random(),
            CustomerIdMother.random(),
            AccountStatus.OPEN,
            DateValueObjectMother.random(),
            [DebitMother.withThisValue(value)],
            []
        );

    }
}

File: python_django/core_test/account/domain/account_mother.py
from core.account.domain.account import Account
from core.customer.domain.customer_id import CustomerId
from core_test.account.domain.accont_id_mother import AccountIdMother


class AccountMother(object):
    @staticmethod
    def with_zero_balance_and_customer_id(customer_id: CustomerId) -> Account:
        account_id = AccountIdMother.random()
        return Account(account_id, customer_id, 0)
